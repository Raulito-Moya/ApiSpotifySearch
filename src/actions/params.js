import { types } from "../types/types"



 export const setParams = (artist, type, market) => ({
   
    type: types.params,
    payload: {
        artist: artist,
        type: type,
        market: market
    }

})
