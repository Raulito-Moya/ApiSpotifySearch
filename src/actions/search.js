import { fetchconToken } from "../helpers/fetch"
import { types } from "../types/types";


export const search = (token, q, type, market) => {
   return async(dispatch) => {
   
    const res = await fetchconToken(token,q,type,market);
      
     const {data} = await res
     console.log(data);
      if(data.albums){

          const {items} = data.albums
        // console.log(items);
         dispatch(getItems(items))

      }else if (data.tracks) {

         const {items} = data.tracks
         console.log(items);
         dispatch(getItems(items))
      }
   }
   


}


export const getItems = (items) => ({
  
   type: types.items,
   payload: items

})