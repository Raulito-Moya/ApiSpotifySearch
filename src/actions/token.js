import { fetchAuthUser, fetchconToken, fetchToken } from "../helpers/fetch"
import { types } from "../types/types"


export const getAuthorize = async() => {
   const url = await fetchAuthUser()

   console.log(url.data);
   window.location.assign(url.data)
}


 export const getFetchToken = () => {  //esto no todavia
    return async(dispatch) => {

    const  res = await fetchToken();
     const {token} = await res.json();
     console.log(token);
      dispatch(getToken(token))
    }
 }



export const getToken = (token) => ({
   
    type: types.token,
    payload: token

})


export const logout = () => ({

   type: types.logout

})