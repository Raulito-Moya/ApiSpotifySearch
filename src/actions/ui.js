import { types } from "../types/types"



 export const errorFormLogin = (data) => ({

   type: types.uierrorForm,
   payload: data

})

export const cleanError = () => ({

  type: types.cleanError

})



