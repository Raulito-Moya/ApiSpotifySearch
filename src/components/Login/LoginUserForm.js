import React from 'react'
import { useForm } from '../../Hooks/useForm'

export const LoginFormUser = () => {
   
    const [{correo, password}, handleInputChange ] = useForm({
        correo: '',
        password: ''
    })


    const handleLogin = (e) => {
        e.preventDefault()
        console.log(correo, password);
    }
       
  return(

    <div className="row item">
         
      <form type='submit' className="col-sm-10">
          <h2>Login</h2>
          <label className="form-label user">Correo</label>
          <input className="form-control name" type="click" name="correo" value={correo}  onChange={handleInputChange} placeholder="correo"/>

          <label className="form-label user">Password</label>
          <input className="form-control" type="click" name="password" value={password} onChange={handleInputChange}  placeholder="Password"/>
          
          <button type="submit" onClick={handleLogin} className="buttom">LoginUser</button>
       </form>
       
    </div>


  )



}