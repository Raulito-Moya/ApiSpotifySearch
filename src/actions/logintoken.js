import { getUserByID, sendAcountData, sendLogindata } from "../helpers/UserLogin";
import { fetchAuthUser, fetchconToken, fetchToken } from "../helpers/fetch"
import { types } from "../types/types"
import { cleanError, errorFormLogin } from "./ui";
import {getToken} from '../helpers/fetch'

export const getAuthorize = async() => {
   const url = await fetchAuthUser()
   window.location.href = url.data

     
}


export const createAcount= (data) => {
  return async(dispatch) => {
      
      const res = await sendAcountData(data) 
      const body = await res.json()
       console.log(body);
       if(body.msg){
          dispatch(errorFormLogin(body))
       } else{
          dispatch(getUser(body))
        
       }
      
  }

}

export const loginUser = (data) => {
   return async(dispatch) => {

    const res = await sendLogindata(data)
    const body = await res.json()

    if(body.msg){
       dispatch(errorFormLogin(body))
    }else{
       console.log(body);
     
       dispatch(getUser(body))
       dispatch(cleanError())
    }
    


   }
  
}


export const loginUserSpotify = (data) => ({
  
    type: types.login,
    payload: data

})


export const getUserExpecific = (data) => {
  return async(dispatch) => {

    const res = await getUserByID(data)
    const body = await res.json()
    console.log(body);

    dispatch(getUser(body))

  }



}



export const getUser = (data) => ({
   
    type: types.login,
    payload: data

})


export const logout = () => ({

   type: types.logout

})