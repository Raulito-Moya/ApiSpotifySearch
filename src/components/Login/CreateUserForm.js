import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import Swal from 'sweetalert2';

import { useForm } from '../../Hooks/useForm'
import { createAcount } from '../../actions/logintoken'
import { sendAcountData } from '../../helpers/UserLogin'
import { passwordError } from '../../actions/ui'


export const CreateUserForm = () => {

     const dispatch = useDispatch()
     const { auth } = useSelector( state => state )
  
    const [{nombre,correo, password,confirmPassword}, handleInputChange ] = useForm({
        nombre:'',
        correo: '',
        password: '',
        confirmPassword:''
    })


    const handleCreateUser = async(e) => {
 
      try {
        
        console.log(nombre,correo,password);
        e.preventDefault()
           if(password !== confirmPassword ){
             Swal.fire('Error', 'Las contrasenas deben ser igaules','error')
           }else if(nombre.length <= 2){
            Swal.fire('Error', 'El nombre deben ser un nombe valido','error')
           } else if(correo.length <= 7){
             Swal.fire('Error', 'El correo debe ser un correo valido','error')
           } else{
             dispatch(createAcount({nombre,correo,password})) 
             localStorage.setItem('login', JSON.stringify(auth.login))
           }


      } catch (error) {
        console.log(error);
      }

     
             
         
    }




  return(

    <div className="row item">
         
      <form type='submit' className="col-sm-12">
          <h2>Create a Account</h2>
          <label className="form-label user">Name</label>
          <input type="click" name="nombre" value={nombre} onChange={handleInputChange} className="form-control name" placeholder="nombre"/>

          <label className="form-label user">Correo</label>
          <input type="click" name="correo" value={correo} onChange={handleInputChange}  className="form-control" placeholder="name@example.com"/>
          <label className="form-label user">Password</label>
          <input type="click" name="password" value={password} onChange={handleInputChange} className="form-control " placeholder="password"/>
          <input type="click" name="confirmPassword" value={confirmPassword} onChange={handleInputChange} className="form-control " placeholder="Confirm Password"/>


          <button type="submit" onClick={handleCreateUser} className="buttom">LoginUser</button>
       </form>
       
    </div>


  )



}