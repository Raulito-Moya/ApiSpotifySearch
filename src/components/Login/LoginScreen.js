import axios from 'axios';
import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getFetchToken, getAuthorize } from '../../actions/logintoken';
import { CreateUserForm } from './CreateUserForm';
import { LoginFormUser } from './LoginUserForm';

export const LoginScreen = () => {



  const initState = {
    login: true
  }
  const { ui } = useSelector(state => state)   



    const handleLoginSpotify = () => {
      const data = getAuthorize() 
    
    //  console.log(data);
     // dispatch(getFetchToken())
      //localStorage.setItem('login',JSON.stringify(initState))
    }

    


      const [form, setForm] = useState(JSON.parse(localStorage.getItem('formvalue')) )
      let formValue = 'value' //el value para renderear cada formulario

      const setFormStatus = () => { 
        setForm(!form) 
        console.log(form); 
        }

        JSON.stringify(localStorage.setItem('formvalue',form )) //aqui mando esto fuera del setFormSatus ya que si lo mnado dentro no me coge el valor cambioado por el setState
       formValue = JSON.parse(localStorage.getItem('formvalue')) 
       console.log(formValue);
 


    return(
      <div className="row">
        
        <div className="loginScreen">
           
          { formValue && <LoginFormUser/> }

          { !formValue && <CreateUserForm/> } 
         
          {ui.msg && <h5>Usuario o Contrasena incorrectos</h5>}
         <div className="buttom-change">
           <button type="click" className="btn"  onClick={setFormStatus}>{formValue ? 'Create a Acount ':'Login User'}</button>
         </div>
         
          
          <div className="item">
            <h4>Login with Spotify</h4>
            <button 
             type="click"
             onClick={handleLoginSpotify}
            >
              <img src="/img/spotify-logo.png" alt=""/>
            </button>
          </div>

        </div>
      </div>
    )
}