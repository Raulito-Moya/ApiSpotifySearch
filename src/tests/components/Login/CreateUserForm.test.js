import thunk from "redux-thunk";
import configureStore from 'redux-mock-store'

import { CreateUserForm } from "../../../components/Login/CreateUserForm";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { createAcount } from "../../../actions/logintoken";


const middlewares = [thunk]
const mockstore = configureStore( middlewares )
   
jest.mock('../../../actions/logintoken', () => ({
  createAcount: jest.fn()

}))

const initState = {auth:{
   login: true
}}

let store = mockstore(initState)
store.dispatch = jest.fn();




Storage.prototype.setItem = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <CreateUserForm />
    </Provider>
)





describe('CreateUserForm tests',()=> {
      
  beforeEach(()=>{
    jest.clearAllMocks()

  })
  
  test('debe mostrarse correctemente',()=> {

      expect(wrapper).toMatchSnapshot()

    })
  

  test('crear usuario', ()=> {
  
      wrapper.find('input[name="nombre"]').simulate('change', {
        target: {
            name: 'nombre',
            value: 'Juan'
        }
    })
  
    wrapper.find('input[name="correo"]').simulate('change', {
      target: {
          name: 'correo',
          value: 'Juan@gmail.com'
      }
     })
  
  
     wrapper.find('input[name="password"]').simulate('change', {
      target: {
          name: 'password',
          value: '123456'
      }
     })
  
     wrapper.find('input[name="confirmPassword"]').simulate('change', {
      target: {
          name: 'confirmPassword',
          value: '123456'
      }
     })
  
     wrapper.find('button[className="buttom"]').simulate('click')
  
  
      
      expect(createAcount).toHaveBeenCalled()
   
   expect( localStorage.setItem ).toHaveBeenCalledWith('login','true')   //to do me da un  error en aqui en la prueba

  })  
  

})

