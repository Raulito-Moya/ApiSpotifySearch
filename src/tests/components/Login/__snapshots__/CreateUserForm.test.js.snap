// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserForm tests debe mostrarse correctemente 1`] = `
<div
  className="row item"
>
  <form
    className="col-sm-12"
    type="submit"
  >
    <h2>
      Create a Account
    </h2>
    <label
      className="form-label user"
    >
      Name
    </label>
    <input
      className="form-control name"
      name="nombre"
      onChange={[Function]}
      placeholder="nombre"
      type="click"
      value=""
    />
    <label
      className="form-label user"
    >
      Correo
    </label>
    <input
      className="form-control"
      name="correo"
      onChange={[Function]}
      placeholder="name@example.com"
      type="click"
      value=""
    />
    <label
      className="form-label user"
    >
      Password
    </label>
    <input
      className="form-control "
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="click"
      value=""
    />
    <input
      className="form-control "
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="click"
      value=""
    />
    <button
      className="buttom"
      onClick={[Function]}
      type="submit"
    >
      LoginUser
    </button>
  </form>
</div>
`;
